### PR TITLE
V0.6.2

### PR DESCRIPTION
**Pull request by SC426**
y-py 0.6.2

**Destination channel:** defaults

### Links

- [PKG-3801](https://anaconda.atlassian.net/browse/PKG-3801)
- [Upstream repository](https://github.com/y-crdt/ypy/tree/v0.6.2)
- [Upstream changelog/diff](https://github.com/y-crdt/ypy/releases)
- Relevant dependency PRs:
  - https://github.com/AnacondaRecipes/rust-activation-feedstock/pull/6
  - https://github.com/AnacondaRecipes/rust-feedstock/pull/7

### Explanation of changes:

- Update version
- Dependency updates
- Add osx x86_64 compiler restraints https://github.com/rust-lang/rust/pull/104385

[PKG-3801]: https://anaconda.atlassian.net/browse/PKG-3801?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -19,20 +19,26 @@ build:
     - python
 
 requirements:
+  build:
+    run_exports: # [osx and x86_64]
```
Comment (cbouss) [inline recipe/meta.yaml]: run_exports should go under the main build section and not the one under requirements.

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  - 1.76.0
+# use {{ compiler('rust-gnu') }} when requiring a build using the m2w64-toolchain
+rust_gnu_compiler:             # [win]
+  - rust-gnu                   # [win]
+rust_gnu_compiler_version:     # [win]
+  - 1.76.0                     # [win]
+MACOSX_DEPLOYMENT_TARGET:   # [osx and x86_64]
+  - "10.12"                 # [osx and x86_64]
+rust_compiler_version:
+  - 1.76.0
```
Comment (JeanChristopheMorinPerso) [inline recipe/conda_build_config.yaml]:
```suggestion
```
This is already defined at line 3 and 4.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,15 +1,18 @@
...
 
 source:
   url: https://pypi.io/packages/source/y/y_py/y_py-{{ version }}.tar.gz
-  sha256: 50cfa0532bcee27edb8c64743b49570e28bb76a00cd384ead1d84b6f052d9368
+  sha256: 4757a82a50406a0b3a333aa0122019a331bd6f16e49fed67dca423f928b3fd4d
 
 build:
+  run_exports: # [osx and x86_64]
+    strong_constrains: # [osx and x86_64]
+      - __osx >={{ MACOSX_DEPLOYMENT_TARGET|default("10.12") }}  # [osx and x86_64]
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: Do we really need this? You'll get these for free if you use your latest builds for rust and rust-activation if I'm not mistaken.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -21,18 +24,20 @@ build:
...
     - {{ compiler('c') }}
     - {{ compiler('rust') }}
     - cargo-bundle-licenses
   host:
-    - maturin =0.13.7
+    - maturin 1.3.1
     - pip
     - python
     - wheel
+    - setuptools
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
```
setuptools and wheel are not needed when using maturin.

File: recipe/meta.yaml (hunk 3 above)
Comment (Jrice1317) [reply recipe/meta.yaml]: I was just about to ask if this was required as well

File: recipe/meta.yaml [hunk 5]
```diff
@@ -21,18 +24,20 @@ build:
...
   run:
     # this is needed otherwise import y_py in tests fails with ModuleNotFoundError
     - python
+    - __osx >=10.12     # [osx and x86_64]
```
Comment (Jrice1317) [inline recipe/meta.yaml]: Is this needed if the cbc.yaml has 10.12 listed for osx?

**Comment by Jrice1317** [comment]
I noticed rust and rust-activation are both listed in the PR's description, but it seems a lot of what is done in this PR would be solved when rust is uploaded. Is that correct or am I missing something regarding the order of these PRs?

**Comment by SC426** [comment]
> I noticed rust and rust-activation are both listed in the PR's description, but it seems a lot of what is done in this PR would be solved when rust is uploaded. Is that correct or am I missing something regarding the order of these PRs?

They're linked here: https://github.com/AnacondaRecipes/y-py-feedstock/pull/4/files#diff-ff61408cdc05bc9667deeadb55e4aaceb1371972076b6bf6934f9008920f2bd2

**Comment by JeanChristopheMorinPerso** [comment]
https://staging.continuum.io/package/prefect/fs/y-py-feedstock/pr4/7c3f648/osx-64/y-py-0.6.2-py310h7854f77_0.conda shows that the osx-64 packages pins `__osx >=10.9`. This seems wrong to me... Shouldn't it be 10.12?

**Comment by SC426** [comment]
> f13a4ca

**Comment by SC426** [comment]
> https://staging.continuum.io/package/prefect/fs/y-py-feedstock/pr4/7c3f648/osx-64/y-py-0.6.2-py310h7854f77_0.conda shows that the osx-64 packages pins `__osx >=10.9`. This seems wrong to me... Shouldn't it be 10.12?

What was it before I removed the constraints? Tried to check myself but couldn't see.

**Comment by JeanChristopheMorinPerso** [comment]
Actually, that's the constraint, not dep. Though, I would have expected `__osx` to be in the dependencies (in https://staging.continuum.io/package/prefect/fs/y-py-feedstock/pr4/7c3f648/osx-64/y-py-0.6.2-py310h7854f77_0.conda).

**Review by Jrice1317 — APPROVED** [review]
Given everything builds successfully, LGTM!

**Comment by JeanChristopheMorinPerso** [comment]
Ah, I know what that `constraints` thing is. It's basically the `run_constrained`. Anyway, the version of `__osx` injected is wrong.

File: abs.yaml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+channels:
+  - https://staging.continuum.io/prefect/fs/rust-feedstock/pr7/1481b8e
+  - https://staging.continuum.io/prefect/fs/rust-activation-feedstock/pr6/b2372ac
```
Comment (cbouss) [inline abs.yaml]:
```suggestion
  - https://staging.continuum.io/prefect/fs/rust-activation-feedstock/pr6/433790c
```

**Comment by cbouss** [comment]
> Ah, I know what that `constraints` thing is. It's basically the `run_constrained`. Anyway, the version of `__osx` injected is wrong.

It might by the contrains export from the rust activation feedstock not being set because of two build sections.
We should rebuild from the latest commit of rust-activation and check the end result. But for now the osx-64 workers are busy..

**Comment by SC426** [comment]
@JeanChristopheMorinPerso abd @cbouss the latest builds just finished.

**Comment by JeanChristopheMorinPerso** [comment]
Still looks wrong to me: https://staging.continuum.io/package/prefect/fs/y-py-feedstock/pr4/814e88b/osx-64/y-py-0.6.2-py310h7854f77_0.conda.

**Comment by cbouss** [comment]
> 4c5c04d

Now it's fine, we had conda_build_config.yaml misplaced in rust-activation.

**Review by JeanChristopheMorinPerso — APPROVED** [review]

**Review by cbouss — APPROVED** [review]

**Pull request merged**